server/resp/src/main/java/org/infinispan/server/resp/RespServer.java

```diff
@@ -48,7 +48,8 @@ protected void startInternal() {
          } else {
             if (cacheManager.getCacheManagerConfiguration().isClustered()) { // We are running in clustered mode
                builder.clustering().cacheMode(CacheMode.REPL_SYNC);
-               builder.clustering().hash().hashFunction(CRC16.getInstance());
+               // See: https://redis.io/docs/reference/cluster-spec/#key-distribution-model
+               builder.clustering().hash().keyPartitioner(new CRC16HashFunctionPartitioner()).numSegments(16384);
```

Comment:
I wonder should we at least log a WARN message when the user configures the RESP cache to be clustered and not have CRC16 as the key partitioner?

Reply:
I think we should. Let me update it.